rotkehlchen/db/utils.py

```diff
@@ -260,19 +260,29 @@ def deserialize_tags_from_db(val: Optional[str]) -> Optional[list[str]]:
     return tags
 
 
-def _prepare_tag_mappings(
+def _get_tag_reference(
         entry: Union['ManuallyTrackedBalance', BlockchainAccountData, 'XpubData'],
         object_reference_keys: list[
             Literal['id', 'chain', 'address', 'xpub.xpub', 'derivation_path'],
         ],
-) -> list[tuple[str, str]]:
-    """Common function to prepare tag mappings. Caller has to make sure entry.tags is not None"""
+) -> str:
     reference = ''
-    mapping_tuples = []
     for key in object_reference_keys:
         value = rgetattr(entry, key)
         if value is not None:  # value.value is for SupportedBlockchain
             reference += str(value) if key != 'chain' else value.value
+    return reference
+
+
+def _prepare_tag_mappings(
+        entry: Union['ManuallyTrackedBalance', BlockchainAccountData, 'XpubData'],
+        object_reference_keys: list[
+            Literal['id', 'chain', 'address', 'xpub.xpub', 'derivation_path'],
+        ],
+) -> list[tuple[str, str]]:
+    """Common function to prepare tag mappings. Caller has to make sure entry.tags is not None"""
+    mapping_tuples = []
+    reference = _get_tag_reference(entry, object_reference_keys)
```

Comment:
Is there a need for this new function?
It doesn't do anything different from how the previous logic works. Am I missing something?

Edit: I see you use it below.